### PR TITLE
Javascript javascript3 week3

### PR DESCRIPTION
**Pull request by alina-kho**
Homework for JS3 W3 is done. Any comments and suggestions are higly appreciated. Have a nice weekend :)

### CODE REVIEW

File: javascript/javascript3/week3/art/main.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  draw() {
+    ctx.beginPath();
+    ctx.arc(this.x, this.y, this.r, this.startAngle, this.endAngle);
+    ctx.fillStyle = this.fillColor;
+    ctx.fill();
+    ctx.stroke();
+  }
+}
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/art/main.js:36]: To make the class reusable, we can pass "context" as an argument to the draw method. 
  draw(context) {
      context.beginPath();
      context.arc(this.x, this.y, this.r, this.startAngle, this.endAngle);
      context.fillStyle = this.fillColor;
      context.fill();
      context.stroke();
  }

const c1 = new Circle(50, 50, 20, 0, 2 * Math.PI, "#000000");
c1.draw(ctx);

File: javascript/javascript3/week3/art/main.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    `#${Math.floor(Math.random() * 0xffffff).toString(16)}`
+  );
+  circle.draw();
+}
+
+function drawOnInterval() {
+  setInterval(() => {
+    drawRandomCircle();
+  }, 100);
+}
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/art/main.js:61]: superb!!!!:muscle:

File: javascript/javascript3/week3/art/main.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  const circle = new Circle(
+    pX,
+    pY,
+    minRadius + Math.random() * (maxRadius - minRadius),
+    0,
+    2 * Math.PI,
+    `#${Math.floor(Math.random() * 0xffffff).toString(16)}`
+  );
+  circle.draw();
+});
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/art/main.js:80]: Well done!!! :smiley:

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+const billy = new Product("Billy shelf", 550, "DKK");
+shoppingCart.addProduct(billy);
+
+shoppingCart.getUser(3);
+
+// shoppingCart.getTotal();
+flatscreen.convertToCurrency("EUR");
+console.log(flatscreen.price);
+console.log(flatscreen.currency);
+shoppingCart.renderProducts();
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/shopping-cart/main.js:107]: Great job Alina.
Keep it up!!!

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  getTotal() {
+    const total = this.products.reduce(
+      (sum, product) => sum + product.price,
+      0
+    );
+    totalPar.innerHTML = `Total: ${total}`;
+    return total;
+  }
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/shopping-cart/main.js:67]: Great job using reduce 💪

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    this.products = this.products.filter((el) => el !== product);
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/shopping-cart/main.js:51]: You can also do it with "splice".

Comment (alina-kho) [reply javascript/javascript3/week3/shopping-cart/main.js:51]: Thank you for the hint ;)

File: javascript/javascript3/week3/art/main.js (hunk 1 above)
Comment (alina-kho) [reply javascript/javascript3/week3/art/main.js:36]: Fixed! Thank you for the feedback and really cool hints :)

**Pull request merged**